firebase/deploy

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+main() {
+   set -euo pipefail
+
+   if [[ "$#" -gt 1 ]]; then
+       exit_error "usage [CHANNEL=dev]"
+   fi
+
+   channel="dev"
+   if [[ "$#" -eq 1 ]]; then
+       channel="$1"
+   fi
+
+   if [[ -n "${CI:-}" ]]; then
+   	setup_ci_creds
+   fi
+
+   if [[ "${channel}" == "live" ]]; then
+      firebase --config firebase/firebase.json deploy --only hosting
+      exit 0
+   fi
+
+   result=$(firebase --json --config firebase/firebase.json hosting:channel:deploy "${channel}")
+   status=$(jq -r '.status' <<<"${result}")
+
+   if [[ "${status}" == "error" ]]; then
+      error_msg=$(jq -r '.error' <<<"${result}")
+      exit_error "${error_msg}"
+   fi
+   if [[ "${status}" != "success" ]]; then
+    	exit_error "unknown status: ${result}"
+   fi
+
+   url=$(jq -r '.result."evy-lang".url' <<<"${result}")
+   pr_num=$(gh pr view --json number -q .number)
+   body="firebase-deployment: $url (${GITHUB_SHA:-$(git rev-parse HEAD)})"
+   gh pr comment "${pr_num}" --body "${body}"
+}
+
+setup_ci_creds() {
+	if [ -z "${FIREBASE_SERVICE_ACCOUNT}" ]; then
+		exit_error "FIREBASE_SERVICE_ACCOUNT environment variable must be set on CI."
+   fi
+	printenv FIREBASE_SERVICE_ACCOUNT > out/gac.json
+	export GOOGLE_APPLICATION_CREDENTIALS=out/gac.json
+}
+
+exit_error() {
+   echo "$1"
```

Comment:
`echo "$*"` so you can provide multiple arguments: `"$*"` will concatenate them with a space (by default) into a single string.

Reply:
❤️ that's so cool, thanks. done. and used.